funcs/func_hangupcause.c

```diff
@@ -139,25 +141,50 @@ static int hangupcause_read(struct ast_channel *chan, const char *cmd, char *dat
 		return -1;
 	}
 
-	ast_channel_lock(chan);
-	cause_code = ast_channel_dialed_causes_find(chan, args.channel);
-	ast_channel_unlock(chan);
+//	ast_log(LOG_ERROR, "Cause for channel '%s'\n", args.channel);
 
-	if (!cause_code) {
-		ast_log(LOG_WARNING, "Unable to find information for channel %s\n", args.channel);
+	if (!(causelist = ast_str_create(128))) {
+		ast_log(LOG_ERROR, "Unable to allocate dynamic string buffer\n");
```

Comment:
A user will see this, so the error message should be useful and explain what will happen and the channel involved such as:

Unable to allocate cause list storage for <channel name>, cause information will be unavailable